src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1150,14 +1158,17 @@ ApplyExtendedOpNodes(MultiExtendedOp *originalNode, MultiExtendedOp *masterNode,
  * operator node.
  */
 static void
-TransformSubqueryNode(MultiTable *subqueryNode)
+TransformSubqueryNode(MultiTable *subqueryNode, List *tableNodeList)
 {
 	MultiExtendedOp *extendedOpNode =
 		(MultiExtendedOp *) ChildNode((MultiUnaryNode *) subqueryNode);
 	MultiNode *collectNode = ChildNode((MultiUnaryNode *) extendedOpNode);
 	MultiNode *collectChildNode = ChildNode((MultiUnaryNode *) collectNode);
+	bool groupedByDistinctPartitionColumn =
```

Comment:
The value of this ```bool``` is not necessarily same for the main query and sub-queries. If I wanted to pass a ```bool``` to this function, then I had to  do "```MultiExtendedOp *extendedOpNode = (MultiExtendedOp *) ChildNode((MultiUnaryNode *) subqueryNode)```" also before calling this function. I thought passing the ```tableNodeList``` to avoid repeating the ```extendedOpNode = ChildNode(...)``` looked better. What do you think?

Reply:
I agree; I must have misread something to believe that my suggestion was equivalent to what was written. Leave it the way it is.